api-module-library/slack/api.js

```diff
@@ -78,6 +80,28 @@ class Api extends OAuth2Requester {
         this.tokenUri = this.baseUrl + this.URLs.access_token;
     }
 
+    async setTokens(params) {
+        this.access_token = get(params, 'access_token');
+        this.team_access_token = get(params, 'access_token');
+        this.refresh_token = get(params, 'refresh_token', null);
+        const authedUser = get(params, 'authed_user', null);
+        const accessExpiresIn = get(params, 'expires_in', null);
+        const refreshExpiresIn = get(
+            params,
+            'x_refresh_token_expires_in',
+            null
+        );
+
+        if (authedUser && authedUser.access_token) {
+            this.access_token = authedUser.access_token;
+        }
+
+        this.accessTokenExpire = moment().add(accessExpiresIn, 'seconds');
+        this.refreshTokenExpire = moment().add(refreshExpiresIn, 'seconds');
+
+        await this.notify(this.DLGT_TOKEN_UPDATE);
+    }
+
```

Comment:
I'm not sure I fully understand this, but it seems like team_access_token is never really used? 
I think that it be better to override getTokenFromCode, since the job that setToken is doesn't need to change. The check around authedUser can be performed outside the scope of setTokens. 
Actually, maybe I understand that this is done also so that the refresh request does not need the same logic? In that case I might still slightly prefer leaving this lower level method with it's base implementation. 
But also, feel free to ignore.

Reply:
I'm removing `team_access_token`. It was from a different implementation that I discarded later.

Here we are using the user access token if it exists, instead of the team token.

I think it makes more sense to update setTokens since the logic we're changing is exactly when setting the tokens. The rest stays the same.